### ISSUE TITLE
Bind via Kerberos

### ISSUE THREAD

**Issue by luandrea** [ltb-project/ltb-common#51]
As stated in the documentation:
For Kerberos authentication (GSSAPI) comment out comment out $ldap_bind*

I think it should be more correct to put them as null, otherwise you do not enter in the correct if in the bind class ([Ldap.php](https://github.com/ltb-project/ltb-common/blob/main/src/Ltb/Ldap.php)).

I'm testing it with docker and I have to put it like this:
`$ldap_binddn = null;
$ldap_bindpw = null;`

**Comment by luandrea** [comment]
I'm referring to this documentation:
https://self-service-password.readthedocs.io/en/stable/config_ldap.html

It's about self-service-password application, but the source code is here.

**Comment by davidcoutadeur** [comment]
Yes, this is because $ldap_binddn and $ldap_bindpw are declared as default variables in config.inc.php:

```
$ldap_binddn = "cn=manager,dc=example,dc=com";
$ldap_bindpw = 'secret';
```

We should indeed update the doc to explain how to nullify these variables.

Thanks for the report.

**Comment by davidcoutadeur** [comment]
After verification, there is already a tip in ssp doc:

```
You can leave these parameters empty to bind anonymously or using GSSAPI (see below). In this case, the password modification must be done with user’s credentials. But this will not work for password reset.
```

I'll add more details.

**Comment by davidcoutadeur** [comment]
Done in https://github.com/ltb-project/self-service-password/commit/f6ea74a8ea678be54b1594094890605f713c5507

**Comment by davidcoutadeur** [comment]
The fix is insufficient, as when:

$ldap_binddn = "";
$ldap_bindpw = "";

the authentication is still tried with ldap bind.

**Comment by davidcoutadeur** [comment]
I have changed the condition for triggering the ldap simple bind or ldap sasl bind.

Now either with $ldap_binddn = null or $ldap_binddn = "", the ldap simple bind would be skipped.

**Comment by davidcoutadeur** [comment]
See also the fix in the ssp documentation: https://github.com/ltb-project/self-service-password/commit/72b8eb1a585707462c69fee943d1f03f49403e29
